src/services/namex.services.ts

```diff
@@ -1,10 +1,405 @@
+        return data
+      }
+
+      // by here we know there is some text in additionalInfo
+      // but it does not contain the exact msg we must add
+      // so we check if there is a previous request_action message
+      // which no longer matches msg because we are editing
+      let allShortDesc = RequestActions.map(request => `*** ${request.shortDesc} ***`)
+      if (allShortDesc.some(desc => data['additionalInfo'].includes(desc))) {
+        let desc = allShortDesc.find(sd => data['additionalInfo'].includes(sd))
+        data['additionalInfo'] = data['additionalInfo'].replace(desc, msg)
+        return data
+      }
+
+      // if there is no previous request_action message then
+      // we just preserve whatever text there is and append msg
+      data['additionalInfo'] += ` \n\n ${msg}`
+      return data
+    } catch (err) {
+      const msg = await this.handleApiError(err, 'Could not add request action comment')
+      console.error('addRequestActionComment() =', msg) // eslint-disable-line no-console
+      await errorModule.setAppError({ id: 'add-request-action-error', error: msg } as ErrorI)
+      return null
+    }
+  }
+  static async cancelPayment (nrId: number, paymentId: string): Promise<NameRequestPayment> {
+    const paymentResponse: NameRequestPayment = {
+      paymentSuccess: false
+    }
+
+    try {
+      const response = await this.axios.delete(`${this.namexUrl()}/payments/${nrId}/payment/${paymentId}`, {
+        headers: { 'Content-Type': 'application/json' }
+      })
+
+      if (response?.status === OK) {
+        paymentResponse.payment = response.data
+        paymentResponse.httpStatusCode = response.status.toString()
+        paymentResponse.paymentSuccess = true
+      } else {
+        // eslint-disable-next-line no-console
+        console.error('cancelPayment(), status was not 200, response =', response)
+        paymentResponse.httpStatusCode = response.status.toString()
+        paymentResponse.paymentSuccess = false
+      }
+
+      return paymentResponse
+    } catch (err) {
+      const msg = await this.handleApiError(err, 'Could not cancel payment')
+      console.error('cancelPayment() =', msg) // eslint-disable-line no-console
+      await errorModule.setAppError({ id: 'cancel-payment-error', error: msg } as ErrorI)
+      return null
+    }
+  }
+  static async checkinNameRequest (nrId: number, nrState: NrState): Promise<boolean> {
+    try {
+      // Approved or Rejected Name Requests are not checked out due to limited data that is editable.
+      // Return out of checkIn because the NR was never checked out.
+      if (nrState === NrState.APPROVED || nrState === NrState.REJECTED) return true
+
+      const checkedOutBy = sessionStorage.getItem('checkedOutBy')
+      const checkedOutDt = sessionStorage.getItem('checkedOutDt')
+
+      if (checkedOutBy) {
+        await this.axios.patch(`${this.namexUrl()}/namerequests/${nrId}/checkin`, {
```

Comment:
This change needs to be verified (tested).

Ie,
1. Try editing in one tab and see if blocked in another tab.
2. Try editing, saving, and editing again.